SerialPrograms/Source/PokemonSwSh/Programs/RNG/PokemonSwSh_CramomaticRNG.cpp

```diff
@@ -0,0 +1,251 @@
+/*  Cram-o-matic RNG Manipulation
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ * 
+ *  Credit goes to Anubis for discovering how the Cram-o-matic works 
+ *  and for the original code to calculate how many advances are needed 
+ *  to get the wanted balls.
+ * 
+ */
+
+#include "Common/Cpp/PrettyPrint.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "NintendoSwitch/NintendoSwitch_Settings.h"
+#include "Pokemon/Pokemon_Strings.h"
+#include "PokemonSwSh/Commands/PokemonSwSh_Commands_GameEntry.h"
+#include "PokemonSwSh/Inference/PokemonSwSh_SelectionArrowFinder.h"
+#include "PokemonSwSh/Programs/RNG/PokemonSwSh_BasicRNG.h"
+#include "PokemonSwSh/Programs/RNG/PokemonSwSh_CramomaticRNG.h"
+#include "CommonFramework/VideoPipeline/VideoFeed.h"
+
+namespace PokemonAutomation {
+    namespace NintendoSwitch {
+        namespace PokemonSwSh {
+            using namespace Pokemon;
+
+            CramomaticRNG_Descriptor::CramomaticRNG_Descriptor()
+                : SingleSwitchProgramDescriptor(
+                    "PokemonSwSh:CramomaticRNG",
+                    STRING_POKEMON + " SwSh", "Cram-o-matic RNG",
+                    "ComputerControl/blob/master/Wiki/Programs/PokemonSwSh/CramomaticRNG.md",
+                    "Perform RNG manipulation to get rare balls from the Cram-o-matic.",
+                    FeedbackType::REQUIRED, false,
+                    PABotBaseLevel::PABOTBASE_12KB
+                )
+            {}
+
+            CramomaticRNG::CramomaticRNG()
+                : NUM_ITERATIONS(
+                    "<b>Iterations:</b><br>How often should the Cram-o-matic be used. Four apricorns are used per iteration.</b>",
+                    240
+                )
+                , NUM_NPCS(
+                    "<b>NPCs:</b><br>Number of NPCs in the dojo, including " + STRING_POKEMON + ".",
+                    22
+                )
+                , BALL_TYPE(
+                    "<b>Wanted Ball:</b><br>Exact kind depends on the currently selected apricorn.",
+                    {
+                        {BallType::Poke, "poke", "Poke Ball"},
```

Comment:
I don't know much about RNG but could it be flags ? Like user pick sport + safari + apricorn. It stops when the first is met

Reply:
Sport and safari are mutually exclusive; from an RNG perspective, they are the same (safari: only same color apricorns, sport: at least two different colors). 

All other types would work as flags. 
Maybe every other safari/sport ball could be sport if both are chosen? Not sure how to properly communicate this to the user though.

In general, flags are a better solution. I will try to improve it.